engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ZeebeTaskHeadersValidatorTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.deployment.model.validation;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeHeader;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeTaskHeaders;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+import org.junit.jupiter.api.Test;
+
+public class ZeebeTaskHeadersValidatorTest {
+
+  final ZeebeTaskHeadersValidator sut = new ZeebeTaskHeadersValidator();
+
+  @Test
+  public void shouldNotAddErrorAboutDuplicateKeys() {
+    // given
+    final ValidationResultCollector mockValidationResultCollector =
+        mock(ValidationResultCollector.class);
+
+    final ZeebeTaskHeaders element =
+        creatMockTaskHeaders(List.of(new HeaderEntry("testKey", "testValue")));
```

Comment:
🔧 I'd prefer that this case checks no duplicate keys, but that it does have multiple header entries. That way its edge is closer to the other test case. Currently, this case could also succeed simply because the validator doesn't accept multiple header entries but accepts single entries.